lib/api/pins_api.dart

```diff
@@ -28,8 +29,23 @@ class DefaultPinsApi extends PinsApi {
   }
 
   @override
-  Future<List<PinModel>> getPins() async {
-    final response = await _apiClient.get('/pins');
+  Future<List<PinModel>> getPins(
+      {String userId, String boardId, int limit, int offset}) async {
+    // ここもっといい書き方ないかな？
+    var urlWithQuery = '/pins?';
+    if (userId != null) {
+      urlWithQuery += 'user_id=$userId&';
+    }
+    if (boardId != null) {
+      urlWithQuery += 'board_id=$boardId&';
+    }
+    if (limit != null) {
+      urlWithQuery += 'limit=$limit&';
+    }
+    if (offset != null) {
+      urlWithQuery += 'offset=$offset';
+    }
+    final response = await _apiClient.get(urlWithQuery);
```

Comment:
```suggestion
  Future<List<PinModel>> getPins(
      {String userId, String boardId, int limit, int offset}) async {
    // ここもっといい書き方ないかな？
    String domain = 'localhost:8080';
    Map<String, String> query = {};
    
    if (userId != null) {
      query['user_id] = userId;
    }
    if (boardId != null) {
      query['board_id'] = boardId;
    }
    if (limit != null) {
      query['limit'] = limit;
    }
    if (offset != null) {
      query['offset'] = offset;
    }
   var uri = Uri.http(domain, '/pins', query)
```
https://api.flutter.dev/flutter/dart-core/Uri/Uri.https.html
https://api.flutter.dev/flutter/dart-core/Uri/Uri.http.html

デバッグと本番でわけなきゃだけど。

Reply:
https://stackoverflow.com/questions/52148319/what-is-darts-equivalent-to-kotlins-let
https://doitu.info/blog/5ca2142a8dbc7a001af33cf5

dartのnull check方法使ってもいい感じに書く方法わからないけど、すくなくともquery追加部分はこのメソッドを使うのが良さそう